hedera-node/hedera-app/src/main/java/com/hedera/node/app/grpc/GrpcServiceBuilder.java

```diff
@@ -0,0 +1,177 @@
+     */
+    private static final MarshallerSupplier MARSHALLER_SUPPLIER =
+            new MarshallerSupplier() {
+                @Override
+                public <T> MethodDescriptor.Marshaller<T> get(Class<T> clazz) {
+                    //noinspection unchecked
+                    return (MethodDescriptor.Marshaller<T>) NOOP_MARSHALLER;
+                }
+            };
+
+    /** The name of the service we are building. */
+    private final String serviceName;
+
+    /** The {@link IngestWorkflow} to invoke for transaction methods. */
+    private final IngestWorkflow ingestWorkflow;
+
+    /** The {@link QueryWorkflow} to invoke for query methods. */
+    private final QueryWorkflow queryWorkflow;
+
+    /**
+     * The set of transaction method names that need corresponding service method definitions
+     * generated.
+     */
+    private final Set<String> txMethodNames = new HashSet<>();
+
+    /**
+     * The set of query method names that need corresponding service method definitions generated.
+     */
+    private final Set<String> queryMethodNames = new HashSet<>();
+
+    /**
+     * Creates a new builder.
+     *
+     * @param serviceName The name of the service. Cannot be null or blank.
+     * @param ingestWorkflow The workflow to use for handling all transaction ingestion API calls
+     * @param queryWorkflow The workflow to use for handling all queries
+     */
+    public GrpcServiceBuilder(
+            @Nonnull String serviceName,
+            @Nonnull IngestWorkflow ingestWorkflow,
+            @Nonnull QueryWorkflow queryWorkflow) {
+        this.ingestWorkflow = Objects.requireNonNull(ingestWorkflow);
+        this.queryWorkflow = Objects.requireNonNull(queryWorkflow);
+        this.serviceName = Objects.requireNonNull(serviceName);
+        if (serviceName.isBlank()) {
+            throw new IllegalArgumentException("serviceName cannot be blank");
+        }
+    }
+
+    /**
+     * Register the creation of a new gRPC method for handling transactions with the given name.
+     * This call is idempotent.
+     *
+     * @param methodName The name of the transaction method. Cannot be null or blank.
+     * @return A reference to the builder.
+     */
+    public @Nonnull GrpcServiceBuilder transaction(@Nonnull String methodName) {
+        if (Objects.requireNonNull(methodName).isBlank()) {
+            throw new IllegalArgumentException("The gRPC method name cannot be blank");
+        }
+
+        txMethodNames.add(methodName);
+        return this;
+    }
+
+    /**
+     * Register the creation of a new gRPC method for handling queries with the given name. This
+     * call is idempotent.
+     *
+     * @param methodName The name of the query method. Cannot be null or blank.
+     * @return A reference to the builder.
+     */
+    public @Nonnull GrpcServiceBuilder query(@Nonnull String methodName) {
+        if (Objects.requireNonNull(methodName).isBlank()) {
```

Comment:
should we use the helper methods of the platform commons module for such checks? Still do not now if we want to expose such functionality in the platform or recreate it in the services.